src/core/qgsdatasourceuri.cpp

```diff
@@ -248,6 +248,17 @@ QString QgsDataSourceUri::removePassword( const QString &aUri )
     regexp.setPattern( QStringLiteral( "/.*@" ) );
     safeName.replace( regexp, QStringLiteral( "/@" ) );
   }
+  else if ( aUri.contains( QLatin1String( "postgresql:" ) ) )
+  {
+    // postgresql://user:pwd@...
+    regexp.setPattern( QStringLiteral( "/.*@" ) );
+    const QString matched = regexp.match( aUri ).capturedView().toString();
+
+    QString anonymised = matched.split( QStringLiteral( ":" ) )[0];
+    anonymised.append( QStringLiteral( "@" ) );
```

Comment:
This logic considers that both `user` and `pwd` are present in the uri.
If `pwd` is omitted (eg `trust` in `pg_hba.conf`) the result will be `postgresql://user@@...`

Reply:
Yes indeed. It's fixed and I added a unit test accordingly.